tmail-backend/jmap/extensions-rabbitmq/src/main/java/com/linagora/tmail/james/jmap/RabbitMQEmailAddressContactSubscriber.java

```
@@ -0,0 +1,140 @@
+                                                 Sender sender,
+                                                 EmailAddressContactMessageHandler messageHandler,
+                                                 Configuration configuration) {
+        this.rabbitMQConfiguration = RabbitMQConfiguration.from(configuration);
+        this.receiverProvider = receiverProvider;
+        this.messageHandler = messageHandler;
+        this.sender = sender;
+        this.listener = Sinks.many().multicast().directBestEffort();
+    }
+
+    public void start() {
+        sender.declareExchange(ExchangeSpecification.exchange(rabbitMQConfiguration.getExchangeName())).block();
+        sender.declare(QueueSpecification.queue(rabbitMQConfiguration.getQueueName()).durable(DURABLE)).block();
+        sender.bind(BindingSpecification.binding(rabbitMQConfiguration.getExchangeName(), rabbitMQConfiguration.getRoutingKey(), rabbitMQConfiguration.getQueueName())).block();
+        listenQueueHandle = consumeQueue();
+        messageConsume = messageConsume();
+    }
+
+    private Disposable consumeQueue() {
+        return Flux.using(receiverProvider::createReceiver,
+                receiver -> receiver.consumeAutoAck(rabbitMQConfiguration.getQueueName()),
+                Receiver::close)
+            .subscribeOn(Schedulers.elastic())
+            .map(this::toMessage)
+            .handle(publishIfPresent())
+            .subscribe(event -> listener.emitNext(event, (signalType, emission) -> true));
+    }
+
+    private Optional<EmailAddressContactMessage> toMessage(Delivery delivery) {
+        String message = new String(delivery.getBody(), StandardCharsets.UTF_8);
+        try {
+            return Optional.of(EmailAddressContactMessageSerializer.deserializeEmailAddressContactMessageAsJava(message));
+        } catch (Exception e) {
+            LOGGER.error("Unable to deserialize '{}'", message, e);
+            return Optional.empty();
```

IMO it would be better to propagate the error